pandas/core/generic.py

```diff
@@ -8442,6 +8442,9 @@ def last(self: FrameOrSeries, offset) -> FrameOrSeries:
         When having a DataFrame with dates as index, this function can
         select the last few rows based on a date offset.
 
+        Note that for the last() method to work, the DatetimeIndex
```

Comment:
Personal opinion but I'd rather rephrase the paragraph above to make this clear:
```
For a DataFrame with a sorted DatetimeIndex, this function selects the last few rows based on a date offset.
```

Reply:
Yeah this seems to be better, I wasn't thinking of changing the original section initially but if that's okay then I'll change that too thanks :)

Reply:
@arw2019 I did make the changes but I was wondering as was commented in the issue earlier as well do we need to raise an error message as well incase the index is not sorted?

Reply:
> @arw2019 I did make the changes but I was wondering as was commented in the issue earlier as well do we need to raise an error message as well incase the index is not sorted?

It uses binary search under the hood (logarithmic time) so, with no other information vertifying sortedness (linear time) would degrade performance. If sortedness information is stored somehow then I agree we should throw but AFAIK it isn't in general for a DatetimeIndex.

Reply:
ohkk understood thanks 👍